regression/testing_tool.py

```diff
@@ -205,17 +207,21 @@ def __init__(self, tool="esbmc"):
 
     def run(self, test_case: BaseTest):
         """Execute the test case with `executable`"""
-        process = Popen(test_case.generate_run_argument_list(*self.tool),
-                        stdout=PIPE, stderr=PIPE)
+        cmd = test_case.generate_run_argument_list(*self.tool)
+
         try:
-            stdout, stderr = process.communicate(timeout=self.timeout)
-        except:
-            process.stdout.close()
-            process.stderr.close()
-            process.kill()
-            process.wait()
-            return None, None, 0        
-        return stdout, stderr, process.returncode
+            # use subprocess.run because we want to wait for the subprocess to finish
+            p = subprocess.run(cmd, stdout=PIPE, stderr=PIPE, timeout=self.timeout);
+            # get the RSS (resident set size) of the subprocess that just terminated.
+            # Save the output in a tmp.log and then use the command below
+            # to get the total maximum RSS:
+            #   egrep "mem_usage=[0-9]+" tmp.log -o | cut -d'=' -f2 | paste -sd+ - | bc
+            # see https://docs.python.org/3/library/resource.html for more details
+            print("mem_usage={0} kilobytes".format(getrusage(RUSAGE_CHILDREN).ru_maxrss))
```

Comment:
If I'm not mistaken, this is Unix-specific. See the docs: https://docs.python.org/3/library/resource.html

Reply:
On Windows there is `GetProcessMemoryInfo` and `GetProcessTimes` doing similar things, but I don't know the Python equivalent.

Reply:
yep our windows CI doesn't like it :(
```
 Traceback (most recent call last):
  File "D:\a\esbmc\esbmc\regression\testing_tool.py", line 15, in <module>
    from resource import *
ModuleNotFoundError: No module named 'resource'
```

I'm gonna try to find a Pythonic module/API for `GetProcessMemoryInfo` (fingers crossed)

Reply:
If there's none, just do this output on Unix only and leave Windows as it was.

Reply:
sounds like a way to go